alsdkdefs/__init__.py

```diff
@@ -192,22 +198,48 @@ def validate_operation_ids(spec):
 
     def validate_path_parameters(spec):
         for path, path_obj in spec[OpenAPIKeyWord.PATHS].items():
-            def get_params_of_type(params_obj, tp):
-                return map(lambda p: p['name'], filter(lambda p: p['in'] == tp, params_obj))
-
             methods = _get_path_methods(path_obj)
             path_param_vars = re.findall('{(.*?)}', path)
             path_methods_parameters = __list_flatten(
-                map(lambda m: get_params_of_type(path_obj[m].get(OpenAPIKeyWord.PARAMETERS, []), 'path'), methods))
+                map(lambda m: _get_params_of_type(path_obj[m].get(OpenAPIKeyWord.PARAMETERS, []), 'path'), methods))
             for path_param_var in path_param_vars:
                 base_msg = f"Path {path} parameter {path_param_var}"
                 if path_param_var not in path_methods_parameters:
                     raise AlertLogicOpenApiValidationException(f"{base_msg}: Parameter defined in the path is missing"
                                                                f" from parameters section")
 
+    def validate_compound_schemas_type(spec):
+        # alcli related check,
+        # if alcli fails to find a type in compound schema
+        # it will assume parameter is expecting object,
+        # this might be unexpectable condition, even though it is completely valid case -
```

Comment:
I'd suggest wording the comment that the CLI is limited to inferring a common type for an argument.